src/test/java/com/google/jenkins/plugins/computeengine/ComputeEngineCloudIT.java

```diff
@@ -362,6 +369,51 @@ public void testTemplateNoGoogleLabels() throws Exception {
         }
     }
 
+    // Tests that no snapshot is created when we only have successful builds for given node
+    @Test(timeout = 300000, expected = GoogleJsonResponseException.class)
+    public void testNoSnapshotCreated() throws Exception {
+        logOutput.reset();
+
+        ComputeEngineCloud cloud = (ComputeEngineCloud) r.jenkins.clouds.get(0);
+        cloud.addConfiguration(validInstanceConfiguration());
+
+        FreeStyleProject project = r.createFreeStyleProject();
+        Builder step = new Shell("echo works");
+        project.getBuildersList().add(step);
+        project.setAssignedLabel(new LabelAtom(LABEL));
+
+        FreeStyleBuild build = r.buildAndAssertSuccess(project);
+        Node worker = build.getBuiltOn();
+        worker.toComputer().doDoDelete();
+
+        assertFalse(logs(), logs().contains("snapshot"));
+        client.getSnapshot(projectId, worker.getNodeName());
+    }
+
+    // Tests snapshot is created when we have failure builds for given node
+    @Test(timeout = 300000)
+    public void testSnapshotCreated() throws Exception {
+        logOutput.reset();
+
+        ComputeEngineCloud cloud = (ComputeEngineCloud) r.jenkins.clouds.get(0);
+        cloud.addConfiguration(validInstanceConfiguration());
+
+        FreeStyleProject project = r.createFreeStyleProject();
+        Builder step = new Shell("exit 1");
+        project.getBuildersList().add(step);
+        project.setAssignedLabel(new LabelAtom(LABEL));
+
+        FreeStyleBuild build = r.assertBuildStatus(Result.FAILURE, project.scheduleBuild2(0));
+        Node worker = build.getBuiltOn();
+        worker.toComputer().doDoDelete();
+
+        Snapshot createdSnapshot = client.getSnapshot(projectId, worker.getNodeName());
+        assertEquals(logs(), createdSnapshot.getStatus(), "READY");
+
+        //cleanup
+        client.deleteSnapshot(projectId, worker.getNodeName());
```

Comment:
This should probably be in some try catch block to be sure we did it..

Reply:
Not sure why a try catch would ensure we deleted?

Reply:
That after test we dont leave snapshot on GCP

Reply:
done